bdikit/mapping_algorithms/column_mapping/algorithms.py

```diff
@@ -102,3 +114,51 @@ def get_column_type(self, context, labels, m=10, model="gpt-4-turbo-preview"):
         )
         col_type_content = col_type.choices[0].message.content
         return col_type_content.split(";")
+
+
+class ContrastiveLearningAlgorithm(BaseColumnMappingAlgorithm):
+    def __init__(self):
+        model_path = get_cached_model_or_download("cl-reducer-v0.1")
+        self.api = ContrastiveLearningAPI(model_path=model_path, top_k=20)
```

Comment:
Can we set `top_k=1` here since we only need the first match? Also, maybe we should add the parameter `model_name` to the constructor as suggested below.